src/tsit5/atsit5.jl

```diff
@@ -553,3 +553,30 @@ function DiffEqBase.step!(integ::SimpleATsit5Integrator, dt::Real)
         step!(integ)
     end
 end
+
+#######################################################################################
+# reinit!
+#######################################################################################
+using RecursiveArrayTools
+function DiffEqBase.reinit!(integrator::SimpleATsit5Integrator, u0 = integrator.u;
+  t0 = integrator.t0, dt = integrator.dt)
+
+    # Is modifying the `uprev` necessary? We do `u_modified(i, true)`
+    if isinplace(integrator)
+        recursivecopy!(integrator.u,u0)
+        recursivecopy!(integrator.uprev,integrator.u)
+    else
+        integrator.u = u0
+        integrator.uprev = integrator.u
+    end
+    u_modified!(integrator, true)
+
+    integrator.t = t0
+    integrator.tprev = t0
+    integrator.dt = dt
```

Comment:
you change dt without resetting the adaptvity qoldinit. That's what's causing the difference I think.

Reply:
Yeap you were right, now tests pass! :) Please give me a `@` when you have tagged a new release with this!